### PR TITLE
Fix verify-proof, some cleanup and add a minimal demo script that can be extended.

### PR DESCRIPTION
**Pull request by clangenb**
I suggest that we merge your branch into master after this PR has been merged.

### CODE REVIEW

File: cli/src/trusted_base_cli/commands/verify_proof.rs [hunk 1]
```diff
@@ -15,76 +15,49 @@
...
+use sp_runtime::traits::Keccak256;
 
 #[derive(Parser)]
 pub struct VerifyMerkleProofCommand {
-	merkle_root: String,
-	merkle_proof: String,
-	orders_encoded_len: usize,
-	leaf_index: usize,
-	leaf: String,
+	merkle_proof_json: String,
```
Comment (m-yahya) [inline cli/src/trusted_base_cli/commands/verify_proof.rs:27]: Is that possible to run the `verify_proof` from the CLI?
I see the command accepts a `proof` of type `String`.
But when I try to pass the string proof, it results in an error:

```
./integritee-cli trusted --mrenclave FmLvUktzLZcL8BizX694wecB7nAfb95m6YPPzXa6ypKm --direct verify-proof //Alice "{\"root\":\"0xd889ae8b2b8807ac5845cee7bb8db1996e01554f88df2040353ea0a582df392f\",\"proof\":[\"0x70997831bb3b0a5eb10dd2c55972f35bcbe781d014ad53811d31743450587292\",\"0x91cfc94f5515165bb59ff651be8bbf3058865f393d78bb102ff686b1782ce360\",\"0xccfd66ee8c896087af65b4ae3f577f86ef82092168e61cbf7f08d3951ef1852a\",\"0xff00158da139d439fade205829bfd1153da3b32b45789f567c0898d563d2abce\"],\"number_of_leaves\":10,\"leaf_index\":1,\"leaf\":[1,0,0,0,0,0,0,0,1,100,50,48,50,50,45,48,51,45,48,52,84,48,53,58,48,54,58,48,55,43,48,48,58,48,48,28,97,99,116,111,114,95,49,1,0,0,0,0,154,153,153,153,153,153,33,64,113,61,10,215,163,112,205,63]}"
```
Error:
```
error: Found argument '{"root":"0xd889ae8b2b8807ac5845cee7bb8db1996e01554f88df2040353ea0a582df392f","proof":["0x70997831bb3b0a5eb10dd2c55972f35bcbe781d014ad53811d31743450587292","0x91cfc94f5515165bb59ff651be8bbf3058865f393d78bb102ff686b1782ce360","0xccfd66ee8c896087af65b4ae3f577f86ef82092168e61cbf7f08d3951ef1852a","0xff00158da139d439fade205829bfd1153da3b32b45789f567c0898d563d2abce"],"number_of_leaves":10,"leaf_index":1,"leaf":[1,0,0,0,0,0,0,0,1,100,50,48,50,50,45,48,51,45,48,52,84,48,53,58,48,54,58,48,55,43,48,48,58,48,48,28,97,99,116,111,114,95,49,1,0,0,0,0,154,153,153,153,153,153,33,64,113,61,10,215,163,112,205,63]}' which wasn't expected, or isn't valid in this context
```

Comment (clangenb) [reply cli/src/trusted_base_cli/commands/verify_proof.rs:27]: yes because you have two arguments: Alice and the Proof, but you only need to pass one.

Comment (m-yahya) [reply cli/src/trusted_base_cli/commands/verify_proof.rs:27]: `true`.
Sorry, my bad.

Comment (m-yahya) [reply cli/src/trusted_base_cli/commands/verify_proof.rs:27]: `true`.
Sorry, my bad.

**Review by m-yahya — APPROVED** [review]

File: cli/src/trusted_base_cli/commands/pay_as_bid_proof.rs [hunk 2]
```diff
@@ -70,11 +75,10 @@ pub(crate) fn pay_as_bid_proof(
 		Some(value) => {
 			let proof: MerkleProofWithCodec<_, _> =
 				MerkleProofWithCodec::decode(&mut &value[..]).unwrap();
-			Some(proof)
+			proof
```
Comment (m-yahya) [inline cli/src/trusted_base_cli/commands/pay_as_bid_proof.rs:78]: I'm wondering how the `proof` which is a type of `MerkleProofWithCodec` is converted to `String`?
Can you please say a few words about how this conversion is done? thanks

Comment (clangenb) [reply cli/src/trusted_base_cli/commands/pay_as_bid_proof.rs:78]: I explicitly used `serde_json::to_string` above to ensure that we can use `serde_json::from_str` without an error.

Comment (clangenb) [reply cli/src/trusted_base_cli/commands/pay_as_bid_proof.rs:78]: `serde_json::to_string`/`from_str` can only be used if you derive `Serialize`, `Deserialie`.

Comment (m-yahya) [reply cli/src/trusted_base_cli/commands/pay_as_bid_proof.rs:78]: I see you are converting in line 45 in the print statement. 
Then you use the output of the print statement in the `demo_energy_market.sh`.
The `proof` in line 78 is still the original decoded proof value.
is that correct?

Comment (clangenb) [reply cli/src/trusted_base_cli/commands/pay_as_bid_proof.rs:78]: Yes, exactly. You can see in the function signature that we return a `MerkleProofWithCodec` struct.

**Pull request merged**